lm_eval/models/huggingface.py

```diff
@@ -905,7 +906,14 @@ def _collate(x):
             toks = x[1] + x[2]
             return -len(toks), tuple(toks)
 
-        re_ord = Collator(requests, sort_fn=_collate)
+        re_ord = Collator(
+            requests,
+            sort_fn=_collate,
+            group_by="contexts"
+            if self.AUTO_MODEL_CLASS == transformers.AutoModelForCausalLM
+            else None,
+            group_fn=lambda a: a[-2] + a[-1][:-1],
```

Comment:
Could you possibly either define this lambda expression as a function like `_collate` with a docstring, or put a comment describing what operation this performs?